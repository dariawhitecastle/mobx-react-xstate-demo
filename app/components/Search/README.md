<!-- ! This is a generated file. To make changes, edit <Component>.doc.js ! -->
## CreateSearch
A Search component

## Usage

```javascript
import { Search } from 'components/Search';
<Search />
```

## Properties

**placeholder**

Placeholder to be displayed in the text field.

```
string
```

**onSearch**

Search function that will filter the data in the main component based on user input

```
function
```
  