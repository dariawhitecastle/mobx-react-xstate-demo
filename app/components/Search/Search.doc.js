import { describe, PropTypes } from 'react-desc'

export const doc = Search => {
  const DocumentedSearch = describe(Search)
    .description('A Search component')
    .usage(
      `import { Search } from 'components/Search';
<Search />`,
    )

  DocumentedSearch.propTypes = {
    placeholder: PropTypes.string
      .description('Placeholder to be displayed in the text field.')
      .defaultValue(false),
    onSearch: PropTypes.func.description(
      'Search function that will filter the data in the main component based on user input',
    ),
  }

  return DocumentedSearch
}
