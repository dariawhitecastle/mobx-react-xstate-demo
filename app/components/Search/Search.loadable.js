/**
 * Asynchronously loads the component for Search
 */

import React from 'react'
import loadable from 'utils/loadable'
import { LoadingIndicator } from 'components/LoadingIndicator'

export default loadable(() => import('./Search'), {
  fallback: <LoadingIndicator />,
})
