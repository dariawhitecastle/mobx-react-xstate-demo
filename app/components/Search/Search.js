/* eslint-disable react/prop-types */
import React, { useState, useEffect } from 'react'

// Components
import { Search } from 'grommet-icons'
import { Box } from '../Box'
import { TextInput } from '../TextInput'

// Styles
import colors from '../../utils/colors'

// utils
import { doc } from './Search.doc'
import helpers from '../../utils/helpers'

/**
 *
 * Search
 *
 * Displays a search box with icon and placeholder text
 *
 * - When user types in the search box, it triggers the onSearch prop with the search term
 *
 */
const CreateSearch = ({
  onSearch,
  placeholder,
  margin = { horizontal: 'xlarge', vertical: 'large' },
}) => {
  const [searchTerm, setSearchTerm] = useState('')

  const handleSearch = e => {
    setSearchTerm(e.target.value.trim())
  }
  useEffect(() => onSearch(searchTerm), [searchTerm])

  return (
    <Box
      alignSelf="center"
      align="center"
      border={{
        side: 'all',
        color: colors.mediumGrey,
      }}
      direction="row"
      margin={margin}
      pad={{ horizontal: 'small', vertical: 'xsmall' }}
      round="xsmall"
      width="large"
    >
      <Search color={colors.primaryBlue10} />
      <TextInput onChange={handleSearch} type="search" placeholder={placeholder} plain />
    </Box>
  )
}

export default helpers.createWithDoc({
  docFunction: doc,
  component: CreateSearch,
})
