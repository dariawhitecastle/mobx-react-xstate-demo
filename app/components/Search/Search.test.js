/**
*
* Tests for Search
*
* @see
https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
*
*/

import React from 'react'
import { render } from '@testing-library/react'
import { IntlProvider } from 'react-intl'
// import 'jest-dom/extend-expect'; // add some helpful assertions

import Search from './Search'
import { DEFAULT_LOCALE } from '../../i18n'

const renderComponent = (props = {}) => {
  const onSearch = searchTerm => console.info(searchTerm)
  return render(
    <IntlProvider locale={DEFAULT_LOCALE}>
      <Search onSearch={onSearch} placeholder="Type something..." {...props} />
    </IntlProvider>,
  )
}

test('it renders and matches snapshot', () => {
  const { container } = renderComponent()
  expect(container).toMatchSnapshot()
})

test('It does not log to the console', () => {
  const spy = jest.spyOn(global.console, 'log')
  renderComponent()
  expect(spy).not.toHaveBeenCalled()
})
