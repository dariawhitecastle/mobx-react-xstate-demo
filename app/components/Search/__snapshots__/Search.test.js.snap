// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders and matches snapshot 1`] = `
<div>
  <div
    class="StyledBox-sc-13pk1d4-0 exStsB"
  >
    <svg
      aria-label="Search"
      class="StyledIcon-ofa7kd-0 jjEuVO"
      viewBox="0 0 24 24"
    >
      <path
        d="M15,15 L22,22 L15,15 Z M9.5,17 C13.6421356,17 17,13.6421356 17,9.5 C17,5.35786438 13.6421356,2 9.5,2 C5.35786438,2 2,5.35786438 2,9.5 C2,13.6421356 5.35786438,17 9.5,17 Z"
        fill="none"
        stroke="#000"
        stroke-width="2"
      />
    </svg>
    <div
      class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kYQCpm"
    >
      <input
        autocomplete="off"
        class="StyledTextInput-sc-1x30a0s-0 duvNfq"
        placeholder="Type something..."
        type="search"
        value=""
      />
    </div>
  </div>
</div>
`;
